website/docs/r/lb_listener_rule.html.markdown

```diff
@@ -220,10 +240,50 @@ Authentication Request Extra Params Blocks (for `authentication_request_extra_pa
+
 Condition Blocks (for `condition`) support the following:
 
-* `field` - (Required) The name of the field. Must be one of `path-pattern` for path based routing or `host-header` for host based routing.
-* `values` - (Required) The path patterns to match. A maximum of 1 can be defined.
+* `field` - (Required) The type of condition. Valid values are `host-header`, `http-header`, `http-request-method`, `path-pattern`, `query-string` and `source-ip`.
+* `host_header` - (Optional) Host headers to match. Host Header block fields documented below. Required if `field` is `host-header` and `values` is empty.
+* `http_header` - (Optional) HTTP headers to match. HTTP Header block fields documented below. Required if `field` is `http-header`.
+* `http_request_method` - (Optional) HTTP request methods to match. HTTP Request Method block fields documented below. Required if `field` is `http-request-method`.
+* `path_pattern` - (Optional) Path patterns to match. Path Pattern block fields documented below. Required if `field` is `path-pattern` and `values` is empty.
+* `query_string` - (Optional) Query strings to match. Query String block fields documented below. Required if `field` is `query-string`.
+* `source_ip` - (Optional) Source IPs to match. Source IP block fields documented below. Required if `field` is `source-ip`.
+* `values` - (Optional, **DEPRECATED**) List of exactly one pattern to match. Only valid when `field` is `host-header` or `path-pattern`, and the `host_header` and `path_pattern` blocks have not been set.
+
+Host Header Blocks (for `host_header`) support the following:
+
+* `values` - (Required) List of host patterns to match. Only one needs to match for the condition to be satisfied.
+
+HTTP Header Blocks (for `http_header`) support the following:
+
+* `http_header_name` - (Required) Name of HTTP header to search.
+* `values` - (Required) List of header value patterns to match. Only one needs to match for the condition to be satisfied.
+
+HTTP Request Method Blocks (for `http_request_method`) support the following:
+
+* `values` - (Required) List of HTTP request methods to match. Only one needs to match for the condition to be satisfied.
+
+Path Pattern Blocks (for `path_pattern`) support the following:
+
+* `values` - (Required) List of path patterns to match. Only one needs to match for the condition to be satisfied.
+
+Query String Blocks (for `query_string`) support the following:
+
+* `values` - (Required) List of query string pairs or values to match. Query String Value blocks documented below. Only one pair needs to match for the condition to be satisfied.
+
+Query String Value Blocks (for `query_string.values`) support the following:
+
+* `key` - (Optional) Query string key pattern to match.
+* `value` - (Required) Query string value pattern to match.
+
+Source IP Blocks (for `source_ip`) support the following:
+
+* `values` - (Required) List of CIDR notations to match. Only one needs to match for the condition to be satisfied.
```

Comment:
I would enhance with 
```suggestion
* `values` - (Required) List of CIDR notations to match. You can use both IPv4 and IPv6 addresses. Wildcards are not supported. Only one needs to match for the condition to be satisfied.
```

Thoughts?

Reply:
I tried not to directly parrot the current AWS documentation too much, especially for conditions that the provider is not enforcing. The provider allows wildcards but they will currently be rejected by the AWS endpoint. Could they remove the restriction later?

Reply:
Anything can happen, but at the time they do that, a new version will be out, and it will potentially be a new API Endpoint, so safe.
I feel this is important enough to not be omitted though.

Reply:
I've greatly expanded the documentation blocks with extra details from the API docs.